gapis/service/service.proto

```diff
@@ -1487,6 +1488,29 @@ message DeviceValidationResult {
 
   // Path to the perfetto trace file to help debug validation issues.
   string trace_path = 2;
+
+  // The specific type of validator that was used.
+  enum ValidatorType {
+    INVALID_VALIDATOR = 0;
+
+    // Adreno validator checks for specific counters to ensure that the name and
+    // id is consistent with hardcoded values. Also has specific check to make
+    // sure that the value is within expected ranges.
+    ADRENO = 1;
+
+    // Mali validator checks for specific counters to ensure that the name and
+    // id is consistent with hardcoded values. Also has specific check to make
+    // sure that the value is within expected ranges.
+    MALI = 2;
+
+    // The fallback validator that runs a minimal check for the GPU counter,
+    // running a trace for all available GPU counters and checking to see that
+    // at least one value is non-zero. This is consistent with the CTS test in
+    // Android U.
+    GENERIC = 3;
+  }
+
+  ValidatorType validator_type = 4;
```

Comment:
The syntax is a tad bit. confusing, but for protobufs, each field is assigned its "index" rather than its default value. There's also distinct rules for each data type of what the [default value would be](https://developers.google.com/protocol-buffers/docs/proto3#default)

For this, validator_type will be 0 (INVALID_VALIDATOR) by default

Reply:
ah, thought that was a go file, right, sure, '4' is the fieldID here.